autogpt/commands/openapi.py

```diff
@@ -0,0 +1,400 @@
+"""Handles any openapi API spec"""
+
+import importlib
+import inspect
+import json
+import os
+import re
+import sys
+from pathlib import Path
+from typing import Any, List, Optional, Tuple
+from urllib.parse import urlparse
+
+import openapi_python_client
+import requests
+import yaml
+from openapi_python_client import GeneratorError
+from openapi_python_client.cli import Config as OpenAPIConfig
+from pydantic import BaseModel
+
+from autogpt.commands.command import Command, CommandRegistry
+from autogpt.config import Config
+
+OPENAPI_COMMANDS_REGISTRY = str(Path(os.getcwd()) / "openapi_commands.yaml")
```

Comment:
@collijk could you help me out with how to move this to the workspace? this will probably become a direct abstraction